test/server/http_dynamic_delay_filter_integration_test.cc

```diff
@@ -22,134 +24,75 @@ namespace Nighthawk {
-      : HttpIntegrationTest(Envoy::Http::CodecClient::Type::HTTP1, GetParam()),
-        request_headers_({{":method", "GET"}, {":path", "/"}, {":authority", "host"}}),
-        delay_header_string_(Envoy::Http::LowerCaseString("x-envoy-fault-delay-request")) {}
-
-  // We don't override SetUp(): tests in this file will call setup() instead to avoid having to
-  // create a fixture per filter configuration.
-  void setup(const std::string& config) {
-    config_helper_.addFilter(config);
-    HttpIntegrationTest::initialize();
-  }
-
-  // Fetches a response with request-level configuration set in the request header.
-  Envoy::IntegrationStreamDecoderPtr getResponse(absl::string_view request_level_config,
-                                                 bool setup_for_upstream_request = true) {
-    const Envoy::Http::LowerCaseString key("x-nighthawk-test-server-config");
-    Envoy::Http::TestRequestHeaderMapImpl request_headers = request_headers_;
-    request_headers.setCopy(key, request_level_config);
-    return getResponse(request_headers, setup_for_upstream_request);
-  }
-
-  // Fetches a response with the default request headers, expecting the fake upstream to supply
-  // the response.
-  Envoy::IntegrationStreamDecoderPtr getResponse() { return getResponse(request_headers_); }
-
-  // Fetches a response using the provided request headers. When setup_for_upstream_request
-  // is true, the expectation will be that an upstream request will be needed to provide a
-  // response. If it is set to false, the extension is expected to supply the response, and
-  // no upstream request ought to occur.
-  Envoy::IntegrationStreamDecoderPtr
-  getResponse(const Envoy::Http::TestRequestHeaderMapImpl& request_headers,
-              bool setup_for_upstream_request = true) {
-    cleanupUpstreamAndDownstream();
-    codec_client_ = makeHttpConnection(lookupPort("http"));
-    Envoy::IntegrationStreamDecoderPtr response;
-    if (setup_for_upstream_request) {
-      response = sendRequestAndWaitForResponse(request_headers, 0, default_response_headers_, 0);
-    } else {
-      response = codec_client_->makeHeaderOnlyRequest(request_headers);
-      response->waitForEndStream();
-    }
-    return response;
-  }
-
-  const Envoy::Http::TestRequestHeaderMapImpl request_headers_;
-  const Envoy::Http::LowerCaseString delay_header_string_;
+public:
+  HttpDynamicDelayIntegrationTest() : HttpFilterIntegrationTestBase(GetParam()){};
 };
 
 INSTANTIATE_TEST_SUITE_P(IpVersions, HttpDynamicDelayIntegrationTest,
                          testing::ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()));
 
 // Verify expectations with an empty dynamic-delay configuration.
 TEST_P(HttpDynamicDelayIntegrationTest, NoStaticConfiguration) {
-  setup(R"(
+  initializeFilterConfiguration(R"(
 name: dynamic-delay
 typed_config:
   "@type": type.googleapis.com/nighthawk.server.ResponseOptions
 )");
   // Don't send any config request header
```

Comment:
Contxt for a few comments here: If I understand correctly, we're saving some effort by having these all be one test to prevent having to initialize a filter repeatedly (which I assume is slow?).  That makes sense here.

However, I think it would be good if we clarified individual test cases within the TEST_P block a little.
1. "Don't send any config request header" is describing what we are doing - can we add in what behavior we're expecting this to produce?

2.  Can we space out all of the individual test cases separately:

e.g.
```// Comment for test 1
getResponse();
EXPECT_EQ()

// Comment for test 2
...```